include/librealsense2/dds/dds-device.cpp

```diff
@@ -167,6 +167,61 @@ dds_device::foreach_motion_profile( size_t sensor_index,
     return msg->second.num_of_profiles;
 }
 
+void dds_device::sensor_open( size_t sensor_index, const std::vector< rs2_video_stream > & profiles )
+{
+    using namespace librealsense::dds::topics;
+
+    assert( profiles.size() < device::control::MAX_OPEN_PROFILES );
```

Comment:
What will happen on release? we can overrun no?

Reply:
You are right. I will check and throw an exception if needed

Reply:
Done